packages/compass-export-to-language/src/components/editor/editor.jsx

```diff
@@ -17,44 +17,19 @@ import styles from './editor.module.less';
 class Editor extends PureComponent {
   static displayName = 'EditorComponent';
 
-  // input query can be an object(empty query) or a string(an actual query) so
-  // check for any
   static propTypes = {
-    outputLang: PropTypes.string.isRequired,
-    transpiledExpression: PropTypes.string.isRequired,
-    error: PropTypes.string,
-    imports: PropTypes.string.isRequired,
-    showImports: PropTypes.bool.isRequired,
-    isInput: PropTypes.bool, // display input or output
-    from: PropTypes.string.isRequired // filter for query, agg for agg
+    language: PropTypes.string,
+    value: PropTypes.string.isRequired
   };
 
-  // need to be able to stringify and add spaces to prettify the object
-  componentDidMount() {
-    if (this.props.isInput && this.props.from !== '') {
-      this.editor.setValue(jsBeautify(this.props.from));
-      this.editor.clearSelection();
-    }
+  static defaultProps = {
+    language: 'javascript'
   }
 
   componentDidUpdate() {
-    if (!this.props.isInput) {
-      if (this.props.error) {
-        this.editor.setValue('');
-        this.editor.session.setMode('ace/mode/' + this.props.outputLang || 'javascript');
-        this.editor.clearSelection();
-      } else {
-        const output = this.props.showImports && this.props.imports !== '' ?
```

Comment:
There were three copies of this code which I now factored out. Also a bug where copy to clipboard didn't take the showImports and imports props into account.

Reply:
see https://jira.mongodb.org/browse/COMPASS-4572